java-checks/src/test/files/checks/UnusedPrivateMethod.java

```diff
@@ -0,0 +1,128 @@
+package org.sonar.java.checks.targets;
+
+import java.util.stream.IntStream;
+
+public class UnusedPrivateMethod {
+
+  public UnusedPrivateMethod(String s) {
+    init();
+  }
+
+  private void init() {
+  }
+
+  private void writeObject(java.io.ObjectOutputStream out) throws java.io.IOException {
+    // this method should not be considered as dead code, see Serializable contract
+  }
+
+  private void readObject(java.io.ObjectInputStream in) throws java.io.IOException, ClassNotFoundException {
+    // this method should not be considered as dead code, see Serializable contract
+  }
+
+  private Object writeReplace() throws java.io.ObjectStreamException {
+    // this method should not be considered as dead code, see Serializable contract
+    return null;
+  }
+
+  private Object readResolve() throws java.io.ObjectStreamException {
+    // this method should not be considered as dead code, see Serializable contract
+    return null;
+  }
+
+  private void readObjectNoData() throws java.io.ObjectStreamException {
+    // this method should not be considered as dead code, see Serializable contract
+  }
+
+  @SuppressWarnings("unused")
+  private int unusedPrivateMethod() { // Noncompliant {{Remove this unused private "unusedPrivateMethod" method.}}
```

Comment:
@m-g-sonar this was not in SONARJAVA-958, I added SONARJAVA-1179 to the current PR which takes care of that.